apps/aecore/lib/aecore/structures/spend_tx.ex

```diff
@@ -3,43 +3,129 @@ defmodule Aecore.Structures.SpendTx do
   Aecore structure of a transaction data.
   """
 
-  alias Aecore.Structures.SpendTx
+  @behaviour Aecore.Structures.Transaction
 
-  @type t :: %SpendTx{
-    from_acc: binary(),
+  alias Aeutil.Serialization
+  alias Aeutil.Parser
+  alias Aecore.Structures.Account
+
+  @typedoc "Arbitrary structure data of a transaction"
+  @type payload :: %__MODULE__{} | map()
+
+  @typedoc "Reason for the error"
+  @type reason :: String.t()
+
+  @typedoc "Public key of the account"
+  @type pub_key() :: binary()
+
+  @typedoc "Structure that holds specific transaction info in the chainstate"
+  @type subdomain_chainstate() :: map()
```

Comment:
describe `map()` more specific

Reply:
We typed it map(), because it could be any structure that the chainstate may hold, for a specific transaction. For example it can hold the chainstate of an oracle or of a votingTx and we don't know what structure they might need. That's why we described it as a generic map()

Reply:
yes but this should be described in the type, so everybody (including dyalizer) knows this

Reply:
Ok, I see your point now, will change this